### PR TITLE
Added tag 124, positioning method source

### PR DESCRIPTION
**Pull request by hdefazio**

### CODE REVIEW

**Comment by hdefazio** [comment]
@daniel-riehm PTAL

File: arrows/klv/klv_0601.h [hunk 1]
```diff
@@ -225,6 +225,25 @@ KWIVER_ALGO_KLV_EXPORT
 std::ostream&
 operator<<( std::ostream& os, klv_0601_sensor_fov_name value );
 
+// ----------------------------------------------------------------------------
+/// Source of the navigation positioning information.
+enum KWIVER_ALGO_KLV_EXPORT klv_0601_positioning_method_source
+{
+  KLV_0601_POSITIONING_METHOD_SOURCE_ON_BOARD_INERTIAL_NAVIGATION_SYSTEM,
```
Comment (daniel-riehm) [inline arrows/klv/klv_0601.h:232]:
```suggestion
  KLV_0601_POSITIONING_METHOD_SOURCE_ON_BOARD_INS,
```
INS is a common abbreviation, so we can save a few keystrokes

File: arrows/klv/klv_0601.h [hunk 2]
```diff
@@ -225,6 +225,25 @@ KWIVER_ALGO_KLV_EXPORT
...
+  KLV_0601_POSITIONING_METHOD_SOURCE_NAVIC,
+  KLV_0601_POSITIONING_METHOD_SOURCE_GLONASS,
+  KLV_0601_POSITIONING_METHOD_SOURCE_BEIDOU_1,
+  KLV_0601_POSITIONING_METHOD_SOURCE_BEIDOU_2,
+};
+
+// ----------------------------------------------------------------------------
+KWIVER_ALGO_KLV_EXPORT
+std::ostream&
+operator<<( std::ostream& os, klv_0601_positioning_method_source value );
```
Comment (daniel-riehm) [inline arrows/klv/klv_0601.h:245]: Declaration looks fine, but I don't see a definition anywhere

File: arrows/klv/tests/test_klv_0601.cxx [hunk 3]
```diff
@@ -168,7 +168,7 @@ auto const expected_result = klv_local_set{
       KLV_0102_COUNTRY_CODING_METHOD_GENC_THREE_LETTER,
       std::string{ "CAN" }, kv::nullopt, std::string{ "FRA" } } },
   { KLV_0601_NUMBER_OF_NAVSATS_IN_VIEW,        uint64_t{ 7 } },
-  { KLV_0601_POSITIONING_METHOD_SOURCE,        klv_blob{ { 0x03 } } },
+  { KLV_0601_POSITIONING_METHOD_SOURCE,        uint64_t{ 3 } },
```
Comment (daniel-riehm) [inline arrows/klv/tests/test_klv_0601.cxx:171]: I would suggest making this more explicit (using `1 << ENUM | ...`), to match the `KLV_0601_GENERIC_FLAG_DATA` entry on lines 78-81.

**Review by daniel-riehm — CHANGES_REQUESTED** [review]

File: arrows/klv/klv_0601.cxx [hunk 4]
```diff
@@ -1001,6 +1001,23 @@ operator<<( std::ostream& os, klv_0601_sensor_fov_name value )
...
+{
+  static std::string strings[ KLV_0601_POSITIONING_METHOD_SOURCE_ENUM_END + 1 ] = {
+    "On-board INS",
+    "GPS",
+    "Galileo",
+    "QZSS",
+    "NAVIC",
+    "GLONASS",
+    "BeiDou-1",
+    "BeiDou-2" };
```
Comment (daniel-riehm) [inline arrows/klv/klv_0601.cxx:1016]: You'll need one more here at the end: `"Unknown Positioning Method Source Bit"`. The`min` on the next line clamps any numbers outside the valid range to `ENUM_END`, so those will all get that `"Unknown"` string.

File: arrows/klv/klv_0601.h [hunk 5]
```diff
@@ -225,6 +225,26 @@ KWIVER_ALGO_KLV_EXPORT
 std::ostream&
 operator<<( std::ostream& os, klv_0601_sensor_fov_name value );
 
+// ----------------------------------------------------------------------------
+/// Source of the navigation positioning information.
+enum KWIVER_ALGO_KLV_EXPORT klv_0601_positioning_method_source
```
Comment (daniel-riehm) [inline arrows/klv/klv_0601.h:230]: One more thing I missed the first time through:
```suggestion
enum KWIVER_ALGO_KLV_EXPORT klv_0601_positioning_method_source_bit
```
And edit the enums too (e.g. `...SOURCE_BIT_GPS`), to be consistent with `generic_flag_data_bit`, marking these as bit indices and not simple numeric values.

**Review by daniel-riehm — APPROVED** [review]

**Pull request merged**